### PR TITLE
Add support for Ambrosus Network (AMB)

### PR DESCRIPTION
**Pull request by AndrewAR2**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -206,7 +211,7 @@ DEFINES += CHAINID_UPCASE=\"BSC\" CHAINID_COINNAME=\"BNB\" CHAIN_KIND=CHAIN_KIND
 APPNAME = "Binance Smart Chain"
 else
 ifeq ($(filter clean,$(MAKECMDGOALS)),)
-$(error Unsupported CHAIN - use ethereum, ropsten, ethereum_classic, expanse, poa, artis_sigma1, artis_tau1, rsk, rsk_testnet, ubiq, wanchain, kusd, musicoin, pirl, akroma, atheios, callisto, ethersocial, ellaism, ether1, ethergem, gochain, mix, reosc, hpb, tomochain, tobalaba, dexon, volta, ewc, webchain, thundercore, flare, flare_coston, theta)
+$(error Unsupported CHAIN - use ethereum, ropsten, ethereum_classic, expanse, poa, artis_sigma1, artis_tau1, rsk, rsk_testnet, ubiq, wanchain, kusd, musicoin, pirl, akroma, atheios, callisto, ethersocial, ambrosus, ellaism, ether1, ethergem, gochain, mix, reosc, hpb, tomochain, tobalaba, dexon, volta, ewc, webchain, thundercore, flare, flare_coston, theta)
```
Comment (pscott) [inline Makefile:214]: Could you move `ambrosius` to be at the end of the list please? Easier to track for us :)

File: Makefile [hunk 2]
```diff
@@ -377,4 +382,4 @@ include $(BOLOS_SDK)/Makefile.rules
 dep/%.d: %.c Makefile
 
 listvariants:
-	@echo VARIANTS CHAIN ethereum ropsten ethereum_classic expanse poa rsk rsk_testnet ubiq wanchain pirl akroma atheios callisto ethersocial ether1 gochain musicoin ethergem mix ellaism reosc hpb tomochain dexon volta ewc thundercore flare flare_coston theta bsc
+	@echo VARIANTS CHAIN ethereum ropsten ethereum_classic expanse poa rsk rsk_testnet ubiq wanchain pirl akroma atheios callisto ethersocial ether1 gochain musicoin ethergem mix ambrosus ellaism reosc hpb tomochain dexon volta ewc thundercore flare flare_coston theta bsc
```
Comment (pscott) [inline Makefile:385]: Could you move `ambrosius` to be at the end of the list please? Easier to track for us :)

File: Makefile [hunk 3]
```diff
@@ -67,6 +67,11 @@ DEFINES += HAVE_ETH2
 APPNAME = "Eth Ropsten"
 DEFINES_LIB=
 APP_LOAD_FLAGS=--appFlags 0xa40
+else ifeq ($(CHAIN),ambrosus)
+APP_LOAD_PARAMS += --path "44'/60'"
+DEFINES += CHAINID_UPCASE=\"AMBROSUS\" CHAINID_COINNAME=\"AMB\" CHAIN_KIND=CHAIN_KIND_ETHEREUM CHAIN_ID=16718
+APPNAME = "Ambrosus"
+APP_LOAD_FLAGS=--appFlags 0xa40
```
Comment (pscott) [inline Makefile:74]: I don't think you need this flag :)

File: Makefile [hunk 4]
```diff
@@ -67,6 +67,11 @@ DEFINES += HAVE_ETH2
 APPNAME = "Eth Ropsten"
 DEFINES_LIB=
 APP_LOAD_FLAGS=--appFlags 0xa40
+else ifeq ($(CHAIN),ambrosus)
+APP_LOAD_PARAMS += --path "44'/60'"
+DEFINES += CHAINID_UPCASE=\"AMBROSUS\" CHAINID_COINNAME=\"AMB\" CHAIN_KIND=CHAIN_KIND_ETHEREUM CHAIN_ID=16718
```
Comment (pscott) [inline Makefile:72]: I think you meant `CHAIN_KIND=CHAIN_KIND_AMBROSUS` :D

File: Makefile [hunk 5]
```diff
@@ -67,6 +67,11 @@ DEFINES += HAVE_ETH2
 APPNAME = "Eth Ropsten"
 DEFINES_LIB=
 APP_LOAD_FLAGS=--appFlags 0xa40
+else ifeq ($(CHAIN),ambrosus)
+APP_LOAD_PARAMS += --path "44'/60'"
```
Comment (pscott) [inline Makefile:71]: Are you sure you're using this path?

**Review by pscott — CHANGES_REQUESTED** [review]

**Comment by pscott** [comment]
Hi!
Thanks for your pull request. I think you're missing some code in the app itself, mainly:

1) Adding your chainkind here: https://github.com/LedgerHQ/app-ethereum/blob/5951e2c638966aa97d8e44c529c75b24196d660b/src/chainConfig.h#L58

2) Addind a switch for your chainkind just like so: https://github.com/LedgerHQ/app-ethereum/blob/5951e2c638966aa97d8e44c529c75b24196d660b/src/main.c#L257-L259

3) Adding a getter for your token, just like here: https://github.com/LedgerHQ/app-ethereum/blob/5951e2c638966aa97d8e44c529c75b24196d660b/src/main.c#L358-L360

4) Adding your tokens here: https://github.com/LedgerHQ/app-ethereum/blob/5951e2c638966aa97d8e44c529c75b24196d660b/src/tokens.c#L4569

5) Adding a define for your NUM_TOKEN: https://github.com/LedgerHQ/app-ethereum/blob/5951e2c638966aa97d8e44c529c75b24196d660b/src/tokens.h#L99

6) Adding a token_definition: https://github.com/LedgerHQ/app-ethereum/blob/5951e2c638966aa97d8e44c529c75b24196d660b/src/tokens.h#L133

**Comment by AndrewAR2** [comment]
Hello! Thank you for your review!
> 1..6

Is this necessary if our network is a complete clone of ethereum with PoA, like Kovan?

File: Makefile (hunk 1 above)
Comment (AndrewAR2) [reply Makefile]: Done!

File: Makefile (hunk 2 above)
Comment (AndrewAR2) [reply Makefile]: Done!

File: Makefile (hunk 3 above)
Comment (AndrewAR2) [reply Makefile]: Done!

File: Makefile (hunk 5 above)
Comment (AndrewAR2) [reply Makefile:71]: It is correct!

**Comment by pscott** [comment]
> Hello! Thank you for your review!
> 
> > 1..6
> 
> Is this necessary if our network is a complete clone of ethereum with PoA, like Kovan?

I do not understand your question :)
But you can see that ETC, POA etc are all variants too (check out listvariants in the Makefile) so I think the answer is yes? ^^

**Comment by AndrewAR2** [comment]
> I think you're missing some code in the app itself, mainly:

Thank you! Added!

**Pull request closed without merging**